server/channels/api4/outgoing_oauth_connection.go

```diff
@@ -63,7 +63,7 @@ func ensureOutgoingOAuthConnectionInterface(c *Context, where string) (einterfac
 		return nil, false
 	}
 
-	if c.App.OutgoingOAuthConnections() == nil || c.App.License() == nil || c.App.License().SkuShortName != model.LicenseShortSkuEnterprise {
+	if c.App.OutgoingOAuthConnections() == nil || !model.MinimumEnterpriseLicense(c.App.License()) {
```

Comment:
Non-blocking suggestion - we could consider adding some sort of middleware, or new handler - something that would wrap `APISessionRequired` like `APISessionAndLicenseLevelRequired` and then pass the minimum license level to that to run these checks. It would help remove a lot of this boilerplatey checking at the top of API controllers.